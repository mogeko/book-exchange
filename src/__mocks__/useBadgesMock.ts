import * as hooks from "@/lib/hooks/useBadges";
import { faker } from "@faker-js/faker";

const mock = jest.spyOn(hooks, "default");

const useBadgeMock = {
  target: hooks.default,
  returnResult: () => mock.mockImplementation(() => exampleData),
};

export const exampleData: BadgesType = {
  menus: Array.from({ length: 3 }, () => faker.datatype.number(100)),
  user: Array.from({ length: 2 }, () => faker.datatype.number(100)),
};

type BadgesType = ReturnType<typeof hooks.default>;

export default useBadgeMock;
