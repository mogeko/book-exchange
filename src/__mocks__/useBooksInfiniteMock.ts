import * as hooks from "@/lib/hooks/useBooks";
import { faker } from "@faker-js/faker";

const mock = jest.spyOn(hooks, "useBooksInfinite");
const setSizeMock = jest.fn();

const useBooksInfiniteMock = {
  target: hooks.useBooksInfinite,
  returnResult: (pages = 3) => ({
    success: () =>
      mock.mockImplementation((param) =>
        genExampleRes({}, [pages, param!.limit!])
      ),
    error: () =>
      mock.mockImplementation((param) =>
        genExampleRes({ data: undefined, isError: true }, [
          pages,
          param!.limit!,
        ])
      ),
    loading: () =>
      mock.mockImplementation((param) =>
        genExampleRes({ data: undefined, isLoading: true }, [
          pages,
          param!.limit!,
        ])
      ),
  }),
  genExampleRes,
  setSizeMock,
};

export const exampleData: BookTypes = {
  title: faker.word.noun(20),
  cover: faker.image.image(1280, 1910),
  description: faker.lorem.paragraph(10),
  published: faker.date.past().toISOString(),
  publisher: faker.company.companyName(),
  tags: Array.from({ length: faker.datatype.number({ min: 2, max: 8 }) }, () =>
    faker.lorem.words(2)
  ),
  author: faker.name.firstName(),
  rates: faker.datatype.number(100),
  id: faker.datatype.number(100),
};

function genExampleRes(res: Partial<ResType> = {}, [x, y] = [3, 10]): ResType {
  const exampleRes = {
    data: Array.from({ length: x }, () =>
      Array.from({ length: y }, () => exampleData)
    ),
    isValidating: false,
    isError: false,
    isLoading: false,
    mutate: jest.fn(),
    setSize: setSizeMock,
    size: 1,
  };

  return { ...exampleRes, ...res };
}

type ResType = ReturnType<typeof hooks.useBooksInfinite>;
type BookTypes = hooks.BookTypes;

export default useBooksInfiniteMock;
