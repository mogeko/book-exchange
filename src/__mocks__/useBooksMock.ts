import * as hooks from "@/lib/hooks/useBooks";
import { faker } from "@faker-js/faker";

const mock = jest.spyOn(hooks, "default");

const useBooksMock = {
  target: hooks.default,
  returnResult: () => ({
    success: () =>
      mock.mockImplementation((param) => genExampleRes({}, param!.limit)),
    error: () =>
      mock.mockImplementation((param) =>
        genExampleRes({ books: undefined, isError: true }, param!.limit)
      ),
    loading: () =>
      mock.mockImplementation((param) =>
        genExampleRes({ books: undefined, isLoading: true }, param!.limit)
      ),
  }),
  genExampleRes,
};

export const exampleData: BookType = {
  id: `bk${faker.datatype.number(100)}`,
  title: faker.word.noun(20),
  cover: faker.image.image(1280, 1910),
  tags: Array.from({ length: faker.datatype.number({ min: 2, max: 8 }) }, () =>
    faker.lorem.words(2)
  ),
  rates: faker.datatype.number(100),
  mate: {
    author: faker.name.firstName(),
  },
  desc: {
    short_desc: faker.lorem.paragraph(10),
  },
};

function genExampleRes(res: Partial<ResType> = {}, limit = 10): ResType {
  const exampleRes = {
    books: Array.from({ length: limit }, () => exampleData),
    isValidating: false,
    isError: false,
    isLoading: false,
    mutate: jest.fn(),
  };

  return { ...exampleRes, ...res };
}

type ResType = ReturnType<typeof hooks.default>;
type BookType = hooks.BooksType[0];

export default useBooksMock;
