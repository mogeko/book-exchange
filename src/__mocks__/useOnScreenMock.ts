import * as hooks from "@/lib/useOnScreen";

const useOnScreenMock = {
  target: hooks.default,
  visiable: () => jest.spyOn(hooks, "default").mockImplementation(() => true),
  not: {
    visiable: () =>
      jest.spyOn(hooks, "default").mockImplementation(() => false),
  },
};

export default useOnScreenMock;
