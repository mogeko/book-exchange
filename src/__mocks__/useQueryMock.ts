import * as hooks from "@/lib/hooks/useQuery";

const mock = jest.spyOn(hooks, "default");

const useQueryMock = {
  target: hooks.default,
  returnResult: (res = {}) => mock.mockImplementation(() => genExampleRes(res)),
};

export function genExampleRes(res: Partial<ResType> = {}) {
  const exampleRes = {
    data: undefined,
    isValidating: false,
    isError: false,
    isLoading: false,
    mutate: jest.fn(),
  };

  return { ...exampleRes, ...res };
}

type ResType = hooks.QueryReturn;

export default useQueryMock;
