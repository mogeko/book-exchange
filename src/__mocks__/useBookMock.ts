import * as hooks from "@/lib/hooks/useBooks";
import { exampleData as exampleBooksData } from "@/__mocks__/useBooksMock";
import { genExampleRes } from "@/__mocks__/useQueryMock";
import { faker } from "@faker-js/faker";

const mock = jest.spyOn(hooks, "useBook");

const useBookMock = {
  target: hooks.useBook,
  returnResult: () => ({
    success: () =>
      mock.mockImplementation(() => genExampleRes({ data: exampleData })),
    error: () =>
      mock.mockImplementation(() => genExampleRes({ isError: true })),
    loading: () =>
      mock.mockImplementation(() => genExampleRes({ isLoading: true })),
  }),
};

export const exampleData: BookType = {
  ...exampleBooksData,
  mate: {
    ...exampleBooksData.mate,
    publisher: faker.company.companyName(),
    subtitle: faker.lorem.sentence(10),
    language: "English",
    publication_date: faker.date.past().toISOString(),
    isbn: `978-${faker.datatype.number({ min: 1000000000, max: 1999999999 })}`,
    paperback: faker.datatype.number({ min: 100, max: 1000 }),
  },
  desc: {
    text: faker.lorem.paragraph(50),
    is_folded: faker.datatype.boolean(),
  },
  digest: {
    text: faker.lorem.paragraph(50),
    is_folded: faker.datatype.boolean(),
  },
};

type BookType = hooks.BookType;

export default useBookMock;
