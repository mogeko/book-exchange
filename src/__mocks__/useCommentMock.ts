import * as hooks from "@/lib/hooks/useComments";
import { genExampleRes } from "@/__mocks__/useQueryMock";
import { faker } from "@faker-js/faker";

const mock = jest.spyOn(hooks, "default");

const useCommentMock = {
  target: hooks.default,
  returnResult: () => ({
    success: () =>
      mock.mockImplementation(() => genExampleRes({ data: exampleData })),
    error: () =>
      mock.mockImplementation(() => genExampleRes({ isError: true })),
    loading: () =>
      mock.mockImplementation(() => genExampleRes({ isLoading: true })),
  }),
};

export const exampleData: CommentType = {
  cm_id: `cm${faker.datatype.number({ min: 1000000000, max: 100000000000 })}`,
  msg: faker.lorem.paragraph(50),
};

type CommentType = hooks.CommentType;

export default useCommentMock;
