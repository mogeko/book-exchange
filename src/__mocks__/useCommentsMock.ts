import * as hooks from "@/lib/hooks/useComments";
import { genExampleRes } from "@/__mocks__/useQueryMock";
import { faker } from "@faker-js/faker";

const mock = jest.spyOn(hooks, "default");

const useCommentsMock = {
  target: hooks.default,
  returnResult: () => ({
    success: () =>
      mock.mockImplementation((param) =>
        genExampleRes({
          data: Array.from({ length: param?.limit! }, () => exampleData),
        })
      ),
    error: () =>
      mock.mockImplementation(() => genExampleRes({ isError: true })),
    loading: () =>
      mock.mockImplementation(() => genExampleRes({ isLoading: true })),
  }),
};

export const exampleData: CommentsType = {
  id: `cm${faker.datatype.number({ min: 1000000000, max: 100000000000 })}`,
  author_meta: {
    uid: `${faker.datatype.number({ min: 10000, max: 100000 })}`,
    username: faker.internet.userName(),
    email: faker.internet.email(),
    avatar: faker.internet.avatar(),
  },
  meta: {
    short_review: faker.lorem.lines(1),
    rates: faker.datatype.number(100),
    likes: faker.datatype.number(1000),
    dislike: faker.datatype.number(1000),
    created_at: faker.date.past().toISOString(),
    location: faker.address.county(),
  },
  responds: Array.from(
    { length: faker.datatype.number({ min: 1, max: 10 }) },
    () => ({
      id: `cm${faker.datatype.number({ min: 1000000000, max: 100000000000 })}`,
      author_meta: {
        uid: `${faker.datatype.number({ min: 10000, max: 100000 })}`,
        username: faker.internet.userName(),
        email: faker.internet.email(),
        avatar: faker.internet.avatar(),
      },
      meta: {
        likes: faker.datatype.number(1000),
        dislike: faker.datatype.number(1000),
        created_at: faker.date.past().toISOString(),
        location: faker.address.county(),
      },
      belongs_to: `cm${faker.datatype.number({
        min: 1000000000,
        max: 100000000000,
      })}`,
      msg: faker.lorem.lines(1),
      is_folded: faker.datatype.boolean(),
    })
  ),
  belongs_to: `bk${faker.datatype.number(1000)}`,
  msg: faker.lorem.sentence(),
  is_folded: faker.datatype.boolean(),
};

type CommentsType = hooks.CommentsType[0];

export default useCommentsMock;
