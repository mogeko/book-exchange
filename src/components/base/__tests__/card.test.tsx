import "@testing-library/jest-dom";
import { render, screen } from "@/lib/test-utils";
import { exampleData } from "@/__mocks__/useBooksMock";
import Card, { LongCard } from "@/components/base/card";

describe("Card", () => {
  it("renders a book card correctly", () => {
    const { container } = render(<Card {...exampleData} />);

    expect(container.querySelector("figure a")).toHaveAttribute(
      "href",
      `/books/${exampleData.id}`
    );
    expect(container.querySelector("figure img")).toBeInTheDocument();
    expect(container.querySelector("h2")?.textContent).toBe(exampleData.title);
    expect(container.querySelector("p")?.textContent).toBe(
      exampleData.mate.author
    );
  });

  it("snapshot a book card", () => {
    const { container } = render(<Card {...exampleData} />);

    expect(container).toMatchSnapshot();
  });

  it("snapshot a book skeleton card", () => {
    const { container } = render(<Card.Skeleton />);

    expect(container).toMatchSnapshot();
  });

  it("render a long card", () => {
    const { container } = render(<LongCard {...exampleData} />);

    expect(container.querySelector("figure a")).toHaveAttribute(
      "href",
      `/books/${exampleData.id}`
    );
    expect(container.querySelector("figure img")).toBeInTheDocument();
    expect(container.querySelector("h2")?.textContent).toBe(exampleData.title);
    expect(container.querySelector("p")?.textContent).toBe(
      exampleData.mate.author
    );
  });

  it("snapshot a long card", () => {
    const { container } = render(<LongCard {...exampleData} />);

    expect(container).toMatchSnapshot();
  });

  it("snapshot a long card skeleton", () => {
    const { container } = render(<LongCard.Skeleton />);

    expect(container).toMatchSnapshot();
  });
});
