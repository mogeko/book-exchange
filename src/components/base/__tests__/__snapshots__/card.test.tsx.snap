// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card render a long card 1`] = `
<div>
  <div
    class="flex py-4 gap-3"
  >
    <figure
      aria-label="Cover Image"
      class="flex justify-center items-center"
    >
      <a
        class="relative w-28 h-[calc(7rem/2*3)]"
        href="/books/bk90"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            alt="catch"
            data-nimg="fill"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </a>
    </figure>
    <div
      class="flex flex-col gap-2 h-[calc(7rem/2*3)]"
    >
      <div
        class="inline-flex justify-start"
      >
        <a
          class="link no-underline text-primary hover:bg-primary hover:text-primary-content focus:bg-primary-focus focus:text-primary-content"
          href="/books/bk90"
        >
          <h2
            class="truncate"
          >
            catch
          </h2>
        </a>
      </div>
      <p
        class="truncate"
      >
        Gracie
      </p>
      <div
        class="inline-flex items-center text-amber-400"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M463 192H315.9L271.2 58.6C269 52.1 262.9 48 256 48s-13 4.1-15.2 10.6L196.1 192H48c-8.8 0-16 7.2-16 16 0 .9.1 1.9.3 2.7.2 3.5 1.8 7.4 6.7 11.3l120.9 85.2-46.4 134.9c-2.3 6.5 0 13.8 5.5 18 2.9 2.1 5.6 3.9 9 3.9 3.3 0 7.2-1.7 10-3.6l118-84.1 118 84.1c2.8 2 6.7 3.6 10 3.6 3.4 0 6.1-1.7 8.9-3.9 5.6-4.2 7.8-11.4 5.5-18L352 307.2l119.9-86 2.9-2.5c2.6-2.8 5.2-6.6 5.2-10.7 0-8.8-8.2-16-17-16z"
          />
        </svg>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M463 192H315.9L271.2 58.6C269 52.1 262.9 48 256 48s-13 4.1-15.2 10.6L196.1 192H48c-8.8 0-16 7.2-16 16 0 .9.1 1.9.3 2.7.2 3.5 1.8 7.4 6.7 11.3l120.9 85.2-46.4 134.9c-2.3 6.5 0 13.8 5.5 18 2.9 2.1 5.6 3.9 9 3.9 3.3 0 7.2-1.7 10-3.6l118-84.1 118 84.1c2.8 2 6.7 3.6 10 3.6 3.4 0 6.1-1.7 8.9-3.9 5.6-4.2 7.8-11.4 5.5-18L352 307.2l119.9-86 2.9-2.5c2.6-2.8 5.2-6.6 5.2-10.7 0-8.8-8.2-16-17-16z"
          />
        </svg>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M463 192H315.9L271.2 58.6C269 52.1 262.9 48 256 48s-13 4.1-15.2 10.6L196.1 192H48c-8.8 0-16 7.2-16 16 0 .9.1 1.9.3 2.7.2 3.5 1.8 7.4 6.7 11.3l120.9 85.2-46.4 134.9c-2.3 6.5 0 13.8 5.5 18 2.9 2.1 5.6 3.9 9 3.9 3.3 0 7.2-1.7 10-3.6l118-84.1 118 84.1c2.8 2 6.7 3.6 10 3.6 3.4 0 6.1-1.7 8.9-3.9 5.6-4.2 7.8-11.4 5.5-18L352 307.2l119.9-86 2.9-2.5c2.6-2.8 5.2-6.6 5.2-10.7 0-8.8-8.2-16-17-16z"
          />
        </svg>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M463 192H315.9L271.2 58.6C269 52.1 262.9 48 256 48s-13 4.1-15.2 10.6L196.1 192H48c-8.8 0-16 7.2-16 16 0 .9.1 1.9.3 2.7.2 3.5 1.8 7.4 6.7 11.3l120.9 85.2-46.4 134.9c-2.3 6.5 0 13.8 5.5 18 2.9 2.1 5.6 3.9 9 3.9 3.3 0 7.2-1.7 10-3.6l118-84.1 118 84.1c2.8 2 6.7 3.6 10 3.6 3.4 0 6.1-1.7 8.9-3.9 5.6-4.2 7.8-11.4 5.5-18L352 307.2l119.9-86 2.9-2.5c2.6-2.8 5.2-6.6 5.2-10.7 0-8.8-8.2-16-17-16zm-47.4 35.2l-79.8 57.3c-10 7.2-14.2 20.2-10.2 31.8l30.1 87.7c1.3 3.7-2.9 6.8-6.1 4.6l-77.4-55.2-.1-.1c-10-7-16.1-18.3-16.1-30.5V113.3c0-2.2 3.1-2.7 3.8-.6l29.6 88.2c3.8 11.4 14.5 19.1 26.5 19.1h97.3c3.9 0 5.5 5 2.4 7.2z"
          />
        </svg>
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M463 192H315.9L271.2 58.6C269 52.1 262.9 48 256 48s-13 4.1-15.2 10.6L196.1 192H48c-8.8 0-16 7.2-16 16 0 .9.1 1.9.3 2.7.2 3.5 1.8 7.4 6.7 11.3l120.9 85.2-46.4 134.9c-2.3 6.5 0 13.8 5.5 18 2.9 2.1 5.6 3.9 9 3.9 3.3 0 7.2-1.7 10-3.6l118-84.1 118 84.1c2.8 2 6.7 3.6 10 3.6 3.4 0 6.1-1.7 8.9-3.9 5.6-4.2 7.8-11.4 5.5-18L352 307.2l119.9-86 2.9-2.5c2.6-2.8 5.2-6.6 5.2-10.7 0-8.8-8.2-16-17-16zm-127.2 92.5c-10 7.2-14.2 20.2-10.2 31.8l30.1 87.7c1.3 3.7-2.9 6.8-6.1 4.6l-77.4-55.2c-4.9-3.5-10.6-5.2-16.3-5.2-5.7 0-11.4 1.7-16.2 5.2l-77.4 55.1c-3.2 2.3-7.4-.9-6.1-4.6l30.1-87.7c4-11.8-.2-24.8-10.3-32l-81-57.1c-3.2-2.2-1.6-7.3 2.3-7.3H196c12 0 22.7-7.7 26.5-19.1l29.6-88.2c1.2-3.6 6.4-3.6 7.6 0l29.6 88.2c3.8 11.4 14.5 19.1 26.5 19.1h97.3c3.9 0 5.5 5 2.3 7.2l-79.6 57.5z"
          />
        </svg>
        <span
          class="text-base ml-2"
        >
          7.8
        </span>
      </div>
      <p
        class="line-clamp-3 text-ellipsis"
      >
        Quis dolore similique accusantium officiis adipisci. Dolorum cum et totam eveniet culpa qui quibusdam ad facere. Rerum distinctio atque consequatur amet iure sequi voluptas dolorem. Suscipit non quam. Deserunt omnis sint similique cum aliquid est sunt odio nemo. Inventore est ut ut magnam ea dolores ipsum ea recusandae. Et facere cumque quae fuga dolor ipsa dolor. Unde sit cupiditate. Sint aspernatur ullam a est repellendus ducimus. Maxime id est dolor nobis fuga.
      </p>
    </div>
  </div>
</div>
`;

exports[`Card renders a book card correctly 1`] = `
<div>
  <div
    class="relative flex flex-col w-28"
  >
    <figure
      aria-label="Cover Image"
      class="flex justify-center items-center"
    >
      <a
        class="relative w-28 h-[calc(7rem/2*3)]"
        href="/books/bk90"
      >
        <span
          style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
        >
          <img
            alt="catch"
            data-nimg="fill"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </a>
    </figure>
    <div
      class="flex flex-col text-sm"
    >
      <a
        class="link no-underline text-primary hover:bg-primary hover:text-primary-content focus:bg-primary-focus focus:text-primary-content"
        href="/books/bk90"
      >
        <h2
          class="truncate"
        >
          catch
        </h2>
      </a>
      <p
        class="truncate"
      >
        Gracie
      </p>
    </div>
  </div>
</div>
`;

exports[`Card snapshot a book skeleton card 1`] = `
<div>
  <div
    class="relative flex flex-col mb-4 w-28 animate-pulse"
  >
    <div
      class="mb-2 w-28 h-[calc(7rem/2*3)] bg-slate-700 rounded"
    />
    <div
      class="mb-2 w-24 h-2 bg-slate-700 rounded"
    />
    <div
      class="mb-2 w-20 h-2 bg-slate-700 rounded"
    />
  </div>
</div>
`;

exports[`Card snapshot a long card skeleton 1`] = `
<div>
  <div
    class="relative flex mb-4 w-full gap-3 animate-pulse"
  >
    <div
      class="mb-2 w-28 h-[calc(7rem/2*3)] bg-slate-700 rounded"
    />
    <div
      class="flex flex-col w-full gap-2"
    >
      <div
        class="mb-2 w-1/5 h-2 bg-slate-700 rounded"
      />
      <div
        class="mb-2 w-1/6 h-2 bg-slate-700 rounded"
      />
      <div>
        <div
          class="mb-2 w-4/5 h-2 bg-slate-700 rounded"
        />
        <div
          class="mb-2 w-4/5 h-2 bg-slate-700 rounded"
        />
        <div
          class="mb-2 w-4/5 h-2 bg-slate-700 rounded"
        />
        <div
          class="mb-2 w-4/6 h-2 bg-slate-700 rounded"
        />
      </div>
    </div>
  </div>
</div>
`;
